provider/datagen/src/databake.rs

```diff
@@ -262,37 +259,143 @@ impl DataExporter for BakedDataExporter {
         }
 
-        move_out!(self.mod_files)
+        let mut mod_files = move_out!(self.mod_files).into_inner().expect("poison");
+        for (_, mods) in mod_files.iter_mut() {
+            mods.sort();
+        }
+
+        let mods = mod_files
+            .remove(&PathBuf::new())
+            .expect("root exists")
+            .into_iter()
+            .dedup()
+            .map(|p| p.parse::<TokenStream>().unwrap());
+
+        let mut marker_data_feature = move_out!(self.marker_data_feature)
             .into_inner()
-            .expect("poison")
-            .into_par_iter()
-            .try_for_each(|(path, mut mods)| {
-                mods.sort();
-                let mods = mods
-                    .into_iter()
-                    .dedup()
-                    .map(|p| p.parse::<TokenStream>().unwrap());
-                let maybe_decl = if path.as_os_str().is_empty() {
-                    quote! {
+            .expect("poison");
+        marker_data_feature.sort();
+        let marker_data_feature = marker_data_feature
+            .into_iter()
+            .map(|(marker_str, data_str, feature_str)| {
+                (
+                    marker_str.parse::<TokenStream>().unwrap(),
+                    data_str.parse::<TokenStream>().unwrap(),
+                    feature_str.parse::<TokenStream>().unwrap(),
+                )
+            })
+            .collect::<Vec<_>>();
+
+        let any_cases = marker_data_feature.iter().map(|(marker, data, feature)| {
+            if marker.to_string() == ":: icu_datetime :: provider :: calendar :: DateSkeletonPatternsV1Marker" {
+                quote! {
+                    #feature
+                    <#marker as ::icu_provider::ResourceMarker>::KEY =>
+                        ::icu_provider::AnyPayload::from_rc_payload::<#marker>(
+                            alloc::rc::Rc::new(
+                                ::icu_provider::DataPayload::from_owned(
+                                    ::icu_provider::zerofrom::ZeroFrom::zero_from(
+                                        litemap_slice_get(#data::DATA, key, req)?
+                        )))),
+                }
+            } else {
+                quote!{
+                    #feature
+                    <#marker as ::icu_provider::ResourceMarker>::KEY =>
+                        ::icu_provider::AnyPayload::from_static_ref::<<#marker as ::icu_provider::DataMarker>::Yokeable>(litemap_slice_get(#data::DATA, key, req)?),
+                }
+            }
+        });
+
+        let resource_impls = marker_data_feature.iter().map(|(marker, data, feature)| {
+            quote! {
+                #feature
+                impl_resource_provider!(#marker, #data::DATA);
+            }
+        });
+
+        self.write_to_file(
+            PathBuf::from("mod"),
+            quote! {
+                #(
+                    mod #mods;
+                )*
+
+                macro_rules! declare_baked_provider {
```

Comment:
Question: Why did you make a macro-inside-a-macro instead of just expanding the code here?

If it's because you want to make the `AnyProvider` impl be optional, then can we make that a flag on `datagen` instead?

Reply:
I think being able to datagen a cfgable crate is somewhat valuable. But you can also use a module for that.

Reply:
I think it's nicer to have the impl explicit in the code that includes the baked files, and to not have too much variation in the baked output.